app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/channels/DefaultChannelAdapter.kt

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.navigationoverlay.channels
+
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import org.mozilla.tv.firefox.R
+
+class DefaultChannelAdapter(
+    private val loadUrl: (String) -> Unit,
+    private val onTileLongClick: ((ChannelTile) -> Unit)?,
+    private val onTileFocused: (() -> Unit)?
+) : RecyclerView.Adapter<DefaultChannelTileViewHolder>() {
+
+    private var tiles: List<ChannelTile> = emptyList()
+
+    fun setTiles(newTiles: List<ChannelTile>) {
+        if (itemCount == 0) {
+            this.tiles = newTiles
+            notifyDataSetChanged()
+            return
+        }
+
+        // DiffUtil allows diff calculation between two lists and output a list of update
+        // operations that converts the first list into the second one
+        val diff = DiffUtil.calculateDiff(object : DiffUtil.Callback() {
+            override fun getOldListSize(): Int = tiles.size
+
+            override fun getNewListSize(): Int = newTiles.size
+
+            override fun areItemsTheSame(oldPos: Int, newPos: Int): Boolean {
+                return tiles[oldPos].url == newTiles[newPos].url
```

Comment:
Are urls really considered unique ids? I feel like we should just do the same thing as `areContentsTheSame` to be safe.